### ISSUE TITLE
Configurable way to prevent emulated builds

### ISSUE THREAD

**Issue by brandonSc** [earthly/earthly#3184]
**Use case**
Sometimes builds can be accidentally run using emulation (qemu) when users would have the build fail instead. Running a build with emulation by mistake can be costly when users are billed for compute minutes since the build can take significantly longer.

**Expected Behavior**
Earthly should have a mode that will trigger a failure when emulation needs to be used. It might make sense to extend the `--strict` flag to control this.

**Comment by vladaionescu** [comment]
I believe that we want to keep `--strict` for other cases, like builds not being consistent. Perhaps we can put this under a separate flag, like `--no-emulation`.

**Comment by brandonSc** [comment]
Good idea @alexcb we could do that as a feature flag in satellites without any changes in earthly-core. One downside is that it prevents you from using emulation entirely though, whereas a flag provided to earthly at build time could be set only in CI scripts.

**Comment by idodod** [comment]
@brandonSc should we close this issue as complete?
